src/vivarium_testing_utils/automated_validation/interface.py

```diff
@@ -13,71 +15,79 @@
-        return self._data_loader.artifact_keys()
+        return self._data_loader.get_artifact_keys()
 
     def get_raw_dataset(self, dataset_key: str, source: str) -> pd.DataFrame:
         """Return a copy of the dataset for manual inspection."""
         return self._data_loader.get_dataset(dataset_key, DataSource.from_str(source))
 
-    def upload_custom_data(self, dataset_key: str, data: pd.DataFrame | pd.Series) -> None:
+    def upload_custom_data(
+        self, dataset_key: str, data: pd.DataFrame | pd.Series[float]
+    ) -> None:
         """Upload a custom DataFrame or Series to the context given by a dataset key."""
+        if isinstance(data, pd.Series):
+            data = data.to_frame(name="value")
         self._data_loader.upload_custom_data(dataset_key, data)
 
     def add_comparison(
-        self, measure_key: str, test_source: str, ref_source: str, stratifications: list[str]
+        self,
+        measure_key: str,
+        test_source: str,
+        ref_source: str,
+        stratifications: list[str] = [],
     ) -> None:
         """Add a comparison to the context given a measure key and data sources."""
-        entity_type, entity, measure = measure_key.split(".")
-        measure = MEASURE_KEY_MAPPINGS[entity_type][measure](entity)
+        entity_type, entity, measure_name = measure_key.split(".")
+        measure = MEASURE_KEY_MAPPINGS[entity_type][measure_name](entity)
 
-        test_source = DataSource.from_str(test_source)
+        test_source_enum = DataSource.from_str(test_source)
 
-        if not test_source == DataSource.SIM:
+        if not test_source_enum == DataSource.SIM:
             raise NotImplementedError(
                 f"Fuzzy Comparison for {test_source} source not implemented. Must be SIM."
             )
-        test_raw_datasets = self._get_raw_datasets_from_source(measure, test_source)
+        test_raw_datasets = self._get_raw_datasets_from_source(measure, test_source_enum)
         test_data = measure.get_ratio_data_from_sim(
             **test_raw_datasets,
         )
 
-        ref_source = DataSource.from_str(ref_source)
-        ref_raw_datasets = self._get_raw_datasets_from_source(measure, ref_source)
-        ref_data = measure.get_measure_data(ref_source, **ref_raw_datasets)
+        ref_source_enum = DataSource.from_str(ref_source)
+        ref_raw_datasets = self._get_raw_datasets_from_source(measure, ref_source_enum)
+        ref_data = measure.get_measure_data(ref_source_enum, **ref_raw_datasets)
         comparison = FuzzyComparison(
             measure,
             test_data,
             ref_data,
             stratifications,
         )
-        self.comparisons.update({measure_key: comparison})
+        self.comparisons[measure_key] = comparison
 
-    def verify(self, comparison_key: str, stratifications: list[str] = []):
+    def verify(self, comparison_key: str, stratifications: list[str] = []):  # type: ignore[no-untyped-def]
```

Comment:
These are ignored because comparisons and plotting haven't been implemented, right?